### PR TITLE
Add qiskit circuit tools

### PR DESCRIPTION
**Pull request by eendebakpt**
* Add method to integrate a circuit into a circuit with more qubits.

### CODE REVIEW

File: src/qtt/qiskit/circuits.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+) -> QuantumCircuit:
+    """ Integrate circuit at specified qubits in a larger qubit system
+
+    This can be used for example to integrate a single-qubit experiment in
+    a 5-qubit circuit to be executed on a 5-qubit device.
+
+    Args:
+        qc: QuantumCircuit to be integrated
+        target_qubits: List of qubits to map the circuit on
+        target_control_bits: If None, then use the registers allocated to the target qubits
```
Comment (QFer) [inline src/qtt/qiskit/circuits.py:21]: These are not control bits but classical bits.

File: src/qtt/qiskit/circuits.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    output_qc = QuantumCircuit(number_of_qubits, number_of_qubits)
+
+    qbits = [output_qc.qregs[0][i] for i in target_qubits]
+    if target_control_bits is None:
+        cbits = [output_qc.cregs[0][i] for i in target_qubits]
+    else:
+        cbits = target_control_bits
+    output_qc = output_qc.compose(qc, qubits=qbits, clbits=cbits)
+
+    if add_measurements:
```
Comment (QFer) [inline src/qtt/qiskit/circuits.py:41]: This is a strange option/addition for integration of a circuit, especially when combined more often. Why not keep adding the measurements separate.

File: src/qtt/qiskit/circuits.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+def integrate_circuit(
+    qc: QuantumCircuit,
+    target_qubits: List[int],
+    target_control_bits: Optional[List[int]] = None,
+    number_of_qubits: int = 5,
+    add_measurements: bool = True,
+) -> QuantumCircuit:
+    """ Integrate circuit at specified qubits in a larger qubit system
+
+    This can be used for example to integrate a single-qubit experiment in
```
Comment (QFer) [inline src/qtt/qiskit/circuits.py:15]: I doubt if this is always logical functionality. When the original circuit has 2-qubit gates and only one of them appears in target_qubits what happens then?

File: src/qtt/qiskit/circuits.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        raise Exception(
+            f'number of qubits {qc.num_qubits} in the specified circuit cannot exceed specfied number of qubits {number_of_qubits} in the target system')
+
+    output_qc = QuantumCircuit(number_of_qubits, number_of_qubits)
+
+    qbits = [output_qc.qregs[0][i] for i in target_qubits]
+    if target_control_bits is None:
+        cbits = [output_qc.cregs[0][i] for i in target_qubits]
+    else:
+        cbits = target_control_bits
```
Comment (QFer) [inline src/qtt/qiskit/circuits.py:38]: shouldn't this be cbits = [output_qc.qregs[0][i] for i in target_control_bits]

File: src/qtt/qiskit/circuits.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+from typing import List, Optional
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+
+def integrate_circuit(
+    qc: QuantumCircuit,
```
Comment (QFer) [inline src/qtt/qiskit/circuits.py:7]: What happens when qc exists of more than 1 groups of qregs?

**Review by QFer — CHANGES_REQUESTED** [review]

File: src/qtt/qiskit/circuits.py (hunk 3 above)
Comment (eendebakpt) [reply src/qtt/qiskit/circuits.py:15]: That is not how the `target_qubits` argument works. The 2-qubit gate in a 3-qubit circuit has qubit indices 1, 2 in the original circuit and the target qubits are 1,3, 4 then the 2-qubit gate is applied on the resulting circuit on qubits 3, 4.

File: src/qtt/qiskit/circuits.py (hunk 2 above)
Comment (eendebakpt) [reply src/qtt/qiskit/circuits.py:41]: If the original circuit has measurements, then in the constructed circuit there are only measurements on the `target_qubits`. A common operation is then to add measurements to the remaining qubits.

File: src/qtt/qiskit/circuits.py (hunk 5 above)
Comment (eendebakpt) [reply src/qtt/qiskit/circuits.py:7]: That is no problem. I added a test case to be sure

**Review by QFer — APPROVED** [review]

**Pull request merged**